### PR TITLE
Guardbot, secscanner, secbot, sechud suspect evaluation rework, firearm permit split into firearm and contraband permits.

### PR DESCRIPTION
**Pull request by Gerhazo**
## About the PR 
Splits the firearm permit into two: a firearm permit and a handling of contraband permit.
Reworks guardbuddies', secbots', secscanners', sechuds' assessments of suspects.
Adds some additional logic to support the forementioned as well as optimizations to already existing code.

Review of the code is welcome as it is some additional logic to stuff that runs fairly often, so any possible optimizations as well as correction of mistakes is great.

I've tested the code for errors and having fixed any issues that propped up functionality-wise it seems to work as supposed to, albeit there are many possible cases and I'll continue making sure that there aren't apparent issues, along with potential expansion and/or fixes based on anything that's pointed out/suggested.

## Why's this needed? 
All of the forementioned used to only care about the ID of the people they were examining. Now, if the person's identity isn't obscured, they prioritize their real identity and only rely on ID cards if the target can't be identified by face.

In addition, if their targets had firearm permits, the assessment function used to just flat out agree on the person not being suspicious, no matter what. This meant that whether they were set to wanted in records, were carrying multiple contraband or otherwise they were seen as always fine. 

Now, the firearm permit has been split into a firearm permit and a contraband permit. Suspicious items are evaluated separately on being a firearm or not and depending on possessed permits their contraband value may or may not be used in evaluation of a person.

The secscanner also now displays the person's visible name, this means that a format in example of "PersonName (as OtherName)", further reinforcing that suspects need to be sneakier and hide their identity, else they risk their cover in stolen ID cards being blown.

The summary of these changes means that you can't just evade being detected by bots through carrying any single additional ID (even a blank one from an ID box) without also hiding your face and also having a gun permit doesn't flat-out protect you from any bot scanning anymore either.

The divide into a firearm and contraband permit also allows giving selective permissions to just certain kind of items, for example letting someone handle a taser but not batons or energy swords without being tagged as a suspect.

## Changelog

```
(u)Gerhazo:
(*)Added a "handling of contraband permit" access level present on appropriate jobs' ID cards and identification computers. Firearm permit now only permits you to carry firearms, other kinds of contraband apply under this new permit, which scanners and bots will take into account.
(*)Security sechud glasses, securitrons, guardbuddies and security scanners now prioritise a person's real identity rather than just the ID card they're holding, if the target's face is not concealed.
```

### CODE REVIEW

**Comment by Ley-bunned** [comment]
By baton do you mean sec baton or the riot baton from surplus/spy thief?

File: _std/_accessLevels.dm [hunk 1]
```diff
@@ -42,46 +42,47 @@
...
+#define access_contrabandpermit 40 // Are allowed to handle contraband as far as guardbuddies and secbots are concerned.
+#define access_engineering 41 // General engineering area and substations.
+#define access_engineering_storage 42 // Main metal/tool storage things.
+#define access_engineering_eva 43 // Engineering space suits. Currently unused.
+#define access_engineering_power 44 // APCs and related supplies.
+#define access_engineering_engine 45 // Engine room.
+#define access_engineering_mechanic 46 // Electronics lab.
+#define access_engineering_atmos 47 // Engineering's supply of gas canisters.
+#define access_engineering_control 49 // Engine control room.
+#define access_engineering_chief 50 // CE's office.
```
Comment (pali6) [inline _std/_accessLevels.dm:54]: I don't like shifting other access ids unnecessarily. For example this will break saved custom jobs.

**Comment by Gerhazo** [comment]
> By baton do you mean sec baton or the riot baton from surplus/spy thief?

It'd be both. Whatever's currently normally counted as contraband if not carried by people with appropriate permission (currently it's a firearm permit, which is mildly weird).

File: _std/_accessLevels.dm (hunk 1 above)
Comment (Gerhazo) [reply _std/_accessLevels.dm]: Good point, reverted the shift, picked next available number.

File: code/mob/living/carbon/human/procs/Life.dm [hunk 2]
```diff
@@ -310,18 +310,43 @@
...
+							if (istype(r_hand, /obj/item/gun/))
+								if(!has_carry_permit)
+									contrabandLevel += r_hand.contraband
+							else
+								if(!has_contraband_permit)
+									contrabandLevel += r_hand.contraband
+
+						if (!contrabandLevel && !has_contraband_permit)
+							if (belt)
+								contrabandLevel += belt.contraband
```
Comment (Tarmunora) [inline code/mob/living/carbon/human/procs/Life.dm:338]: Shouldn't a firearms permit allow you to carry a gun on your belt safely?

Comment (Tarmunora) [reply code/mob/living/carbon/human/procs/Life.dm:338]: Applies to all other contraband checks

Comment (Gerhazo) [reply code/mob/living/carbon/human/procs/Life.dm:338]: I only accounted for the wrestling belt as a belt contraband item, forgot about weapons being possible to carry there. Will correct.

**Pull request merged**